lib/src/Base/IterativeStat/IterativeMoments.cxx

```diff
@@ -0,0 +1,276 @@
+Sample IterativeMoments::getCenteredMoments() const
+{
+  return centeredMoments_;
+}
+
+/* Accessor to the maximum order declared */
+UnsignedInteger IterativeMoments::getOrder() const
+{
+  return orderMax_;
+}
+
+/* Accessor to the mean */
+Point IterativeMoments::getMean() const
+{
+  LOGDEBUG(OSS() << "IterativeMoments::getMean()");
+  if (!(iteration_ > 0)) throw InternalException(HERE) << "Error: cannot compute the mean per component of an empty sample.";
+  return centeredMoments_[0] / iteration_;
+}
+
+/* Accessor to the unbiased variance */
+Point IterativeMoments::getVariance() const
+{
+  LOGDEBUG(OSS() << "IterativeMoments::getVariance()");
+  if (!(iteration_ > 0)) throw InternalException(HERE) << "Error: cannot compute the variance per component of an empty sample.";
+  if (!(orderMax_ >= 2)) throw InternalException(HERE) << "Error: unavailable method, the declared maximum should be at least 2.";
+
+  /* Special case for a size 1 */
+  if (iteration_ == 1) return Point(dimension_, 0.0);
+
+  return centeredMoments_[1] / (iteration_ - 1);
+}
+
+/* Accessor to the unbiased estimator of the skewness */
+Point IterativeMoments::getSkewness() const
+{
+  LOGDEBUG(OSS() << "IterativeMoments::getSkewness()");
+  if (!(iteration_ >= 2)) throw InternalException(HERE) << "Error: cannot compute the skewness per component of a sample of size less than 2.";
+  if (!(orderMax_ >= 3)) throw InternalException(HERE) << "Error: unavailable method, the declared maximum order should be at least 3.";
+
+  /* Special case for a size 2 */
+  if (iteration_ == 2) return Point(dimension_, 0.0);
+  
+  Point result(dimension_);
+  const Point varianceEstimator(getVariance());
+  for(UnsignedInteger d = 0; d < dimension_; ++d)
+  {
+    if (!(varianceEstimator[d] < 0.0 || varianceEstimator[d] > 0.0)) throw NotDefinedException(HERE) << "Error: the sample has component " << d << " constant. The skewness is not defined.";
+    result[d] = iteration_ / ((iteration_ - 1.0) * (iteration_ - 2.0)) * centeredMoments_(2, d) / std::pow(varianceEstimator[d], 1.5) ;
+  }
+  return result;
+}
+
+/* Accessor to the unbiased estimator of the kurtosis */
+Point IterativeMoments::getKurtosis() const
+{
+  LOGDEBUG(OSS() << "IterativeMoments::getKurtosis()");
+  if (!(iteration_ >= 3)) throw InternalException(HERE) << "Error: cannot compute the kurtosis per component of a sample of size less than 4.";
+  if (!(orderMax_ >= 4)) throw InternalException(HERE) << "Error: unavailable method, the declared maximum order is lower than 4.";
+
+  /* Special case for a size 3 */
+  if (iteration_ == 3) return Point(dimension_, 0.0);
+
+  Point result(dimension_);
+  const Point varianceEstimator(getVariance());
+  const Point skewnessEstimator(getSkewness());
+  const UnsignedInteger n = iteration_;
+
+  const Scalar factor1 = n * (n + 1.0) / ((n - 1.0) * (n - 2.0) * (n - 3.0));
+  const Scalar factor2 = -3.0 * (3.0 * n - 5) / ((n - 2.0) * (n - 3.0));
```

Comment:
You may add a comment in order to explain the -3.

Reply:
I was not sure of what comment to write, so I suppressed them completely. I looked on the API manual of Sample.computeKurtosis and found the following explanation:

![image](https://user-images.githubusercontent.com/31351465/157127935-84e2be92-ecd3-4531-8725-b5db5392fec1.png)

So I thought that the calculation was unclear and performed the calculation presented at:

https://github.com/openturns/openturns/issues/1929

All in all, we may:

- add a small comment in the C++ like "Unbiased estimator" but that this may suggest that other estimators are biased (but they are unbiased),
- add to the C++ comment a reference to [^Joanes1998]
- create a new theory entry which would explain the maths on iterative statistics with, e.g., a reference to Pebay's paper and other references as well.

In the latter case, I can write such a page, but this can (and should) be done afterwards. 

[^Joanes1998]: Joanes, Derrick N.; Gill, Christine A. (1998), "Comparing measures of sample skewness and kurtosis", _Journal of the Royal Statistical Society, Series D_, **47** (1): 183–189, 

What option do you suggest?

Reply:
The first and third options are good to me. A small comment in the cxx allows for quick comparison with the `Sample` class for developpers.  I left comments with 'unbiased' in the initial classes.
The theory page is a good idea though not related to this specific PR.

Reply:
I updated the comments and added a reference to Joanes's reference. I hope it fits your requirements.

Reply:
That's fine.